BASE/src/Microsoft.ApplicationInsights/Metrics/Extensibility/AutocollectedMetricsExtraction/AutocollectedMetricsExtractor.cs

```diff
@@ -188,7 +194,122 @@ public int MaxDependencyCloudRoleNameValuesToDiscover
+        /// Setting 0 will all values to be replaced with a single value "Other".
+        /// </summary>
+        public int MaxExceptionCloudRoleNameValuesToDiscover
+        {
+            get
+            {
+                return this.extractorForExceptionMetrics.MaxCloudRoleNameValuesToDiscover;
+            }
+
+            set
+            {
+                if (value < 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), value, "MaxExceptionCloudRoleNameValuesToDiscover value may not be negative.");
+                }
+
+                this.extractorForExceptionMetrics.MaxCloudRoleNameValuesToDiscover = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the maximum distinct values for CloudRoleInstance for Trace telemetry.
+        /// Values encountered after this limit is hit will be collapsed into a single value DIMENSION_CAPPED.
+        /// Setting 0 will all values to be replaced with a single value "Other".
+        /// </summary>
+        public int MaxTraceCloudRoleInstanceValuesToDiscover
+        {
+            get
+            {
+                return this.extractorForTraceMetrics.MaxCloudRoleInstanceValuesToDiscover;
+            }
+
+            set
+            {
+                if (value < 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), value, "MaxTraceCloudRoleInstanceValuesToDiscover value may not be negative.");
+                }
+
+                this.extractorForTraceMetrics.MaxCloudRoleInstanceValuesToDiscover = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the maximum distinct values for CloudRoleName for Trace telemetry.
+        /// Values encountered after this limit is hit will be collapsed into a single value DIMENSION_CAPPED.
+        /// Setting 0 will all values to be replaced with a single value "Other".
+        /// </summary>
+        public int MaxTraceCloudRoleNameValuesToDiscover
+        {
+            get
+            {
+                return this.extractorForTraceMetrics.MaxCloudRoleNameValuesToDiscover;
+            }
+
+            set
+            {
+                if (value < 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), value, "MaxTraceCloudRoleNameValuesToDiscover value may not be negative.");
+                }
+
+                this.extractorForTraceMetrics.MaxCloudRoleNameValuesToDiscover = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the maximum distinct values for SeverityLevel for Trace telemetry.
+        /// Values encountered after this limit is hit will be collapsed into a single value DIMENSION_CAPPED.
+        /// Setting 0 will all values to be replaced with a single value "Other".
+        /// </summary>
+        public int MaxTraceSeverityLevelValuesToDiscover
```

Comment:
Curious - how many severity levels do we have (I guess it 5)? Do we want this dimension to be configurable (e.g. who would want to change this to 3 or 10)?

Reply:
It is 5. Ref: https://docs.microsoft.com/en-us/dotnet/api/microsoft.applicationinsights.datacontracts.severitylevel?view=azure-dotnet

Metric schema max allowed is 100, hence I kept this as configurable. Should we consider removing this property?

Reply:
No strong opinion. @cijothomas do you have some general guidance on 1) how do we prioritize the dimension for different scenarios (e.g. it seems severity level could be the most interesting dimension for logs)? 2) what should be the default cap?

Reply:
The MaxValues was introduced as as safety net, to prevents things from going out of control. But in case of Severity, the values are limited already, so it may not required additional safety nets.

Reply:
Removed.